### PR TITLE
Add Script to enroll Yubikey PIV Certificates

### PR DESCRIPTION
**Pull request by plettich**
The script creates a certificate signing request on the Yubikey and sends
it to privacyIDEA to get it signed. It also creates an attestation
certificate to attest the CSR coming from the Yubikey hardware (currently
privacyIDEA does not support this).
This script is still work in progress.

Working on #42 and #43

### CODE REVIEW

File: scripts/privacyidea-enroll-yubikey-piv [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+/_/                       /___/
+   """)
+    click.echo("""
++--- WARNING ------ WARNING ------ WARNING ------ WARNING ------ WARNING ---+
+|                                                                           |
+|                  This script is still work in progress!                   |
+|            Proper Certificate checking can NOT be guaranteed!             |
+|        Currently all (generated) certs will be written out to ease        |
+|                         development and debugging.                        |
+| One day this script might also become part of the privacyIDEA admin tool. |
```
Comment (cornelinux) [inline scripts/privacyidea-enroll-yubikey-piv:319]: I would dump the last sentence.

Comment (plettich) [reply scripts/privacyidea-enroll-yubikey-piv]: Done in ae4395

File: scripts/privacyidea-enroll-yubikey-piv [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+@click.option('-s', '--subject', help='Subject string for the Certificate')
+@click.option('-U', '--url', help='URL of privacyIDEA API.', required=True)
+# TODO: The user is currently given in the Subject String
+@click.option('-u', '--user', help='YubiKey/Certificate user')
+@click.option('-a', '--admin', help='Admin user for the privacyIDEA API')
+@click.option('-p', '--password', help='Password of the admin user')
+@click.option('-t', '--token', help='Admin token for the privacyIDEA API')
+@click.option('-c', '--ca', help='CA connector in privacyIDEA')
+@click.option('-k', '--keyfile', help='Verify response from privacyIDEA with key',
+              default=None)
```
[the remark this answers is not in the document]
Comment (plettich) [reply scripts/privacyidea-enroll-yubikey-piv:195]: Done in 39d2122

File: scripts/privacyidea-enroll-yubikey-piv [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+
+@cli.command('init-cert')
+@click.pass_context
+@click.option('-r', '--reader', help='Use an external smart card reader.',
+              metavar='NAME', default=None)
+@click.option('-d', '--device', type=int, metavar='SERIAL')
+@click_pin_option
+@click_management_key_option
+@click.option('-s', '--subject', help='Subject string for the Certificate')
```
[the remark this answers is not in the document]
Comment (plettich) [reply scripts/privacyidea-enroll-yubikey-piv]: Done in 39d2122

**Comment by plettich** [comment]
>     * Please add required packages to documentation swig libpcsclite-dev
Done in 39d2122
Since these packages are not required by the main privacyidea tool, i have added a check when running the enroll-script.

> 
>     * Please add requirement of a working CAConnector in privacyIDEA to documentation.
Done in 39d2122 

> 
>     * [ ]  Please add documentation for openssl.cnf requirements.
I think this is a privacyIDEA issue since it is currently providing the broken `openssl.cnf`. We should discuss it there.

**Pull request merged**